connector/connect/common/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -33,6 +34,7 @@ message Command {
     CreateScalarFunction create_function = 1;
     WriteOperation write_operation = 2;
     CreateDataFrameViewCommand create_dataframe_view = 3;
+    CommonInlineUserDefinedFunction register_function = 4;
```

Comment:
I would suggest renaming `CommonInlineUserDefinedFunction` to `CommonUserDefinedFunction` since both registered and inline-defined pickled Python UDFs may share the same proto. CC @HyukjinKwon @zhengruifeng @grundprinzip @hvanhovell

Reply:
After double thoughts, I may want to do the renaming in a separate PR for an easier review, considering Scala UDFs also depends on the `CommonInlineUserDefinedFunction`.